cluster/kubernetes/kubernetes.go

```diff
@@ -483,6 +494,114 @@ func (c *Cluster) PublicSSHKey(regenerate bool) (ssh.PublicKey, error) {
 	return publicKey, nil
 }
 
+type servicePod struct {
+	s  v1.Service
+	pc podController
+}
+
+// Internal function to get a collection of service-controller pairs for a given namespace
+func (c *Cluster) allServices(ns string) (serviceControllers []servicePod, _ error) {
```

Comment:
I reckon we could change `AllServices` to use this (and get rid of `makeService`, which would just repeat the matchControllers step). But can be done in a subsequent refactor.